src/Symfony/Component/Translation/Dumper/FileDumper.php

```diff
@@ -82,12 +82,28 @@ public function dump(MessageCatalogue $messages, $options = array())
                 }
             }
             // save file
-            file_put_contents($fullpath, $this->format($messages, $domain));
+            file_put_contents($fullpath, $this->formatCatalogue($messages, $domain, $options));
         }
     }
 
     /**
      * Transforms a domain of a message catalogue to its string representation.
+     * 
+     * This function is provided for backward compatibility.
```

Comment:
this is not a BC layer, given that it is the one child class should implement now.